### PR TITLE
tests.cc-wrapper: Run with an empty path

### PR DESCRIPTION
**Pull request by Artturin**
Ensure our wrappers hardcode all the necessary tools

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Artturin** [comment]
@ofborg build tests.cc-wrapper

**Comment by Artturin** [comment]
https://github.com/ofborg build tests.cc-wrapper

**Comment by Artturin** [comment]
@ofborg build tests.cc-wrapper

**Comment by Artturin** [comment]
@ofborg build tests.cc-wrapper

**Review by matshch — APPROVED** [review]

File: pkgs/test/cc-wrapper/default.nix [hunk 1]
```diff
@@ -20,58 +25,60 @@ in stdenv.mkDerivation {
...
-    NIX_DEBUG=1 $CC -v
-    NIX_DEBUG=1 $CXX -v
+    unset PATH
+
+    NIX_DEBUG=1 ${CC} -v
+    NIX_DEBUG=1 ${CXX} -v
 
     echo "checking whether compiler builds valid C binaries... " >&2
-    $CC -o cc-check ${./cc-main.c}
+    ${CC} -o cc-check ${./cc-main.c}
```
Comment (NickCao) [inline pkgs/test/cc-wrapper/default.nix:34]: Why not `PATH= ${CC} ...`

Comment (Artturin) [reply pkgs/test/cc-wrapper/default.nix:30]: Done

**Comment by Artturin** [comment]
@ofborg build tests.cc-wrapper

**Comment by Artturin** [comment]
Clang tests fail with `/nix/store/0d4v40m12wiplmphh3sh51prkhva2l3g-clang-wrapper-19.1.1/bin/cc: line 254: mktemp: No such file or directory` 
https://github.com/NixOS/nixpkgs/pull/349589 will fix them

**Pull request merged**